internal/supervisor/import.go

```diff
@@ -99,9 +105,96 @@ func NewImportSupervisor(awsClient toolsAWS.AWS, awat AWATClient, store installa
 	}
 }
 
+// completeImports transitions Installations back to stable after
+// checking with the AWAT to make sure that it has detected the
+// completion of the import
+func (s *ImportSupervisor) completeImports() error {
+	installationList, err := s.store.GetInstallations(
+		&model.InstallationFilter{
+			Paging: model.AllPagesNotDeleted(),
+			State:  model.InstallationStateImportComplete,
+		}, false, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to list Installations")
+	}
+
+	for _, installation := range installationList {
+		importStatusList, err := s.awatClient.GetImportStatusesByInstallation(installation.ID)
```

Comment:
Do you think it would make sense to extract the logic inside the for loop into a separate function? This way when we have `defer` with Installation unlock it will unlock immediately after the function completes and does not need to wait for all other installations to finish, which could potentially reduce the time we hold the lock.

Or perhaps we can lock all fetch installations at the beginning just to be safe (see the comment below).